src/main/java/org/prebid/server/bidder/grid/GridBidder.java

```diff
@@ -21,6 +24,28 @@ protected Imp modifyImp(Imp imp, ExtImpGrid impExt) {
         if (impExt.getUid() == null || impExt.getUid() == 0) {
             throw new PreBidException("uid is empty");
         }
+        ExtImp extImp = mapper.mapper().convertValue(
+                imp.getExt(),
+                ExtImp.class
+        );
+
+        if (extImp != null
+                && extImp.getExtImpData() != null
+                && extImp.getExtImpData().getAdServer() != null
+                && StringUtils.isNotEmpty(extImp.getExtImpData().getAdServer().getAdSlot())) {
+
+            ExtImp modifiedExtImp = ExtImp.of(extImp.getPrebid(),
+                    extImp.getBidder(),
+                    extImp.getExtImpData(),
+                    extImp.getExtImpData().getAdServer().getAdSlot());
+
+            ObjectNode modifiedExt = mapper.mapper().valueToTree(
```

Comment:
we can move it directly to imp building
`imp.toBuilder().ext(mapper.mapper().valueToTree(modifiedExtImp)).build()`